packages/api/test/resolvers/post.test.ts

```diff
@@ -0,0 +1,492 @@
+import { expect } from 'chai'
+import { LibraryItem } from '../../src/entity/library_item'
+import { User } from '../../src/entity/user'
+import { deleteLibraryItemById } from '../../src/services/library_item'
+import {
+  createPosts,
+  deletePosts,
+  findPublicPostById,
+} from '../../src/services/post'
+import { findProfile, updateProfile } from '../../src/services/profile'
+import { deleteUser } from '../../src/services/user'
+import { createTestLibraryItem, createTestUser } from '../db'
+import { generateFakeUuid, graphqlRequest, loginAndGetAuthToken } from '../util'
+
+describe('Post Resolvers', () => {
+  let loginUser: User
+  let authToken: string
+
+  before(async () => {
+    // create test user and login
+    loginUser = await createTestUser('loginUser')
+    authToken = await loginAndGetAuthToken(loginUser.email)
+  })
+
+  after(async () => {
+    await deleteUser(loginUser.id)
+  })
+
+  describe('postsResolver', () => {
+    const query = `
+      query Posts($first: Int, $after: String, $userId: ID!) {
+        posts(first: $first, after: $after, userId: $userId) {
+          ... on PostsSuccess {
+            edges {
+              cursor
+              node {
+                id
+                title
+                content
+                ownedByViewer
+              }
+            }
+            pageInfo {
+              startCursor
+              endCursor
+              hasPreviousPage
+              hasNextPage
+            }
+          }
+          ... on PostsError {
+            errorCodes
+          }
+        }
+      }
+    `
+
+    let postIds: Array<string> = []
+    let libraryItem: LibraryItem
+    let libraryItem1: LibraryItem
+
+    before(async () => {
+      libraryItem = await createTestLibraryItem(loginUser.id)
+      libraryItem1 = await createTestLibraryItem(loginUser.id)
+
+      const posts = [
+        {
+          title: 'Post 1',
+          content: 'Content 1',
+          user: loginUser,
+          createdAt: new Date('2021-01-01'),
+          libraryItemIds: [libraryItem.id],
+        },
+        {
+          title: 'Post 2',
+          content: 'Content 2',
+          user: loginUser,
+          createdAt: new Date('2021-01-02'),
+          libraryItemIds: [libraryItem1.id],
+        },
+      ]
+      const newPosts = await createPosts(loginUser.id, posts)
+
+      postIds = newPosts.map((post) => post.id)
+    })
+
+    after(async () => {
+      await deletePosts(loginUser.id, postIds)
+
+      await deleteLibraryItemById(libraryItem.id, loginUser.id)
+      await deleteLibraryItemById(libraryItem1.id, loginUser.id)
+    })
+
+    it('should return an error if the args are invalid', async () => {
+      const response = await graphqlRequest(query, '', {
+        first: 100,
+        userId: loginUser.id,
+      })
+
+      expect(response.body.data.posts.errorCodes).to.eql(['BAD_REQUEST'])
```

Comment:
newbie question. What is the invalid arg here ??  I see a similar request below, testing for an empty array..  Is `100` the invalid arg here ?

```
 it('should return empty array', async () => {
          const response = await graphqlRequest(query, '', {
            first: 10,
            userId: loginUser.id,
          })

          expect(response.body.data.posts.edges).to.be.empty
        })
```

Reply:
Hey @pradeepmurugesan , sorry for the late reply.

I should be more clear here but yes 100 is invalid because the maximum number of posts allowed to requests a time is 50.